bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/core/IBigtableDataClient.java

```diff
@@ -69,7 +69,7 @@
   ApiFuture<Row> readModifyWriteRowAsync(ReadModifyWriteRow readModifyWriteRow);
 
   /** Creates {@link IBulkMutation} batcher. */
-  IBulkMutation createBulkMutationBatcher();
+  IBulkMutation createBulkMutationBatcher(String tableId);
```

Comment:
we should be careful here as this is breaking. Can we deprecate existing functionality and use this and an overloaded method for now?

Reply:
To provide backward compatibility for existing flow, we might need to add an author `IBulkMutation` class (let's say `BulkMutationGCJClientV2`). As the existing wrapper expects `BulkMutationBatcher` and in this change, we are migrating it to `Batcher<>`.
https://github.com/googleapis/java-bigtable-hbase/blob/276834cde845c9a7efee81fb682f5185d5d979b9/bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutationGCJClient.java#L42-L45

I think `IBigtableDataClient` suppose to be an internal interface. If anyone is using that then it would be great for them to migrate to GCJ veneer library instead of using bigtable-core client.

If you feel we should not break this internal API then we can add flow with existing methods.

Reply:
Agreed. These wrappers only exist to abstract the api for the hbase client. All of them should be marked as `@InternalOnly`